### PR TITLE
Dedup on users table for BQ destination

### PR DESCRIPTION
**Pull request by Jayachand**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: warehouse/bigquery/bigquery.go [hunk 1]
```diff
@@ -351,6 +430,14 @@ func (bq *HandleT) LoadUserTables() (errorMap map[string]error) {
 		errorMap[warehouseutils.UsersTable] = status.Err()
 		return
 	}
+
+	err = bq.deleteTable(stagingTableName)
```
Comment (psrikanth88) [inline warehouse/bigquery/bigquery.go:434]: lets move this to defer once its created?

Comment (psrikanth88) [reply warehouse/bigquery/bigquery.go:434]: also do this on crashRecover. Refer to postgres.go for similar stuff

File: warehouse/bigquery/bigquery.go [hunk 2]
```diff
@@ -327,22 +337,91 @@ func (bq *HandleT) LoadUserTables() (errorMap map[string]error) {
 		bqUsersView,                      // 3
 		identifiesFrom,                   // 4
 	)
+	loadUserTable := func() {
```
Comment (psrikanth88) [inline warehouse/bigquery/bigquery.go:340]: rename it to something like loadUserTableByAppend ?

File: warehouse/bigquery/bigquery.go [hunk 3]
```diff
@@ -327,22 +337,91 @@ func (bq *HandleT) LoadUserTables() (errorMap map[string]error) {
...
+			return
+		}
+	}
+
+	if !isUsersTableDedupEnabled {
+		loadUserTable()
+		return
+	}
+
+	stagingTableName := misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, strings.ReplaceAll(uuid.Must(uuid.NewV4()).String(), "-", ""), warehouseutils.UsersTable), 127)
```
Comment (psrikanth88) [inline warehouse/bigquery/bigquery.go:371]: move this also to a func loadUserTableByMerge just for uniformity?

**Review by psrikanth88 — CHANGES_REQUESTED** [review]

File: warehouse/bigquery/bigquery.go (hunk 1 above)
Comment (Jayachand) [reply warehouse/bigquery/bigquery.go]: added

**Review by psrikanth88 — APPROVED** [review]

**Review by sivashanmukh — APPROVED** [review]
LGTM

**Pull request merged**